### ISSUE TITLE
hangs while downloading asar file

### ISSUE THREAD

**Issue by aspiers** [BetterDiscord/Installer#192]
**Describe the bug**

When using v1.0.0-hotfix installer, the install gets stuck on the "Downloading asar file" stage.  My network traffic monitor shows virtually zero traffic on the network interface, so this isn't even a slow download issue like described in #191.

**To Reproduce**
1. Go to https://github.com/BetterDiscord/Installer/releases
2. Download https://github.com/BetterDiscord/Installer/releases/download/v1.0.0-hotfix/BetterDiscord-Linux.AppImage
3. Run it
4. Click through to the install process
5. Observe it hanging on the "Downloading asar file" stage.

**Expected behavior**

The install proceeds and completes in a reasonable timeframe.

**Screenshots**

![image](https://user-images.githubusercontent.com/100738/126717962-e27073d2-8217-4c3a-9a55-079b2bfe2ae9.png)

**System Information:**
- OS: Linux (openSUSE Leap)
- Discord Version: 0.0.15

**Comment by dav1312** [comment]
Duplicate of #35

**Comment by aspiers** [comment]
I agree it looks like it's duplicate, but in practice it's not, because I don't have any antivirus or firewall software which could be blocking this.

**Comment by aspiers** [comment]
I see from #173 that there is an issue with the use of phin for downloads, but #179 indicates that this issue only applies with non-hotfixed installers, whereas I see the problem with the hotfix. (Although weirdly I'm pretty sure the same hotfix installed has worked for me on two other machines running very similar openSUSE Linux distributions.)

I should also mention there's no issue with my network.  It's a fast connection with pure fibre-optic end-to-end, provided by the most reliable ISP in the UK.

**Comment by Elepicos** [comment]
I attempted to do the same today using a reliable internet connection and found the same or a similar error, screenshot attached. Also have no active antivirus. Mine however does not hang, but gives me an error.
![image](https://user-images.githubusercontent.com/73370863/126856340-3d1e6afc-c077-4157-a6d7-20567969cd41.png)

**Comment by dav1312** [comment]
@Elepicos Is clearly **not** the same issue. The issue that you have is already pinned and closed #174

**Comment by tropicaaal** [comment]
Are there any visible errors in console? (You can access DevTools using <kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>I</kbd> with the installer focused).

**Comment by aspiers** [comment]
Nothing relevant, no.  The only message which appears at all is

    DevTools failed to load SourceMap: Could not load content for file:///tmp/.mount_BetterhbVtjk/resources/app.asar/renderer.js.map: 

which appears early on and is pretty obviously unrelated.

**Comment by QbDesu** [comment]
The installer being stuck on asar download is the same issue as before. The hotfix version didn't touch any of that, hence the original issues are still open.

**Comment by aspiers** [comment]
Which original issues are those?  I can't see any in this repo reporting the same thing.

**Comment by dav1312** [comment]
Issue #35 was closed by the user that posted it because it was fixed for him but the issue is still present.
It should still be open until it gets fully fixed in the next installer update because it was already fixed in Phin a few days ago https://github.com/ethanent/centra/pull/12

---

Issue #179 is still open and it mentions 2 issues about Phin:
1. This was already fixed in 384bef6. This is the one that is not a big issue after the hotfix.
    > phin won't follow redirects by default which broke the download after the App repo was moved to the organization. This is a lesser problem, after the hotfix
2. This one is the issue mentioned in #35 
    > Phin has this nasty bug where it will never resolve/reject a promise when the server aborts the connection

**Comment by aspiers** [comment]
Ah, those ones.  As [I mentioned above](https://github.com/BetterDiscord/Installer/issues/192#issuecomment-885308431), #35 is not an exact duplicate of this - even though the symptom is the same problem, it has a different root cause (or at least contributing factor) since it was resolved after the user disabled their firewall and anti-virus.

[I also mentioned above](https://github.com/BetterDiscord/Installer/issues/192#issuecomment-885309985) that #179 claims to apply only to the non-hotfixed installer, which I'm not using.  I'm a little confused whether the `followRedirects` fix in 384bef63e9d6 is present in the non-hotfixed installer, or the hotfixed one, or both.  It seems like an obviously beneficial change with no obvious downsides, so I'm not sure why it wouldn't be present in both.  Is this a possible cause of the hang I am seeing in the hotfixed installer?

Regarding Phin failing to gracefully handle an aborted connection, I can understand that being a problem in some cases, but I don't see why my installer would be aborting the connection 100% of the time, so I'm guessing that is unrelated to my issue.

In summary, even though I'm aware of those known issues, I'm still no closer to knowing how to get past this hang.  Some advice on solving that would be really appreciated.  Thanks!

**Comment by dav1312** [comment]
> [I also mentioned above](https://github.com/BetterDiscord/Installer/issues/192#issuecomment-885309985) that #179 claims to apply only to the non-hotfixed installer, which I'm not using. I'm a little confused whether the `followRedirects` fix in [384bef6](https://github.com/BetterDiscord/Installer/commit/384bef63e9d64a53fb5891a7ba30199645d3c0d7) is present in the non-hotfixed installer, or the hotfixed one, or both. It seems like an obviously beneficial change with no obvious downsides, so I'm not sure why it wouldn't be present in both. Is this a possible cause of the hang I am seeing in the hotfixed installer?

Even tho it claims to apply only to the non-hotfixed installer **neither of the 2 issues mentioned are fixed in the hotfixed version**. 
Is just that the first issue is not a huge problem now in the hotfixed update.

If your connection is fine it could be related to issue #137 

You can also try a [manual installation](https://github.com/BetterDiscord/BetterDiscord#manual-installation)
Or using [betterdiscordctl](https://github.com/bb010g/betterdiscordctl#installation)

**Comment by aspiers** [comment]
Thanks, betterdiscordctl looks like a great option!

**Comment by zerebos** [comment]
I'm hoping to get a build out soon with the updated version of phin to see if this alleviates some of the download-hanging issues

**Comment by ewertonurias** [comment]
> I agree it looks like it's duplicate, but in practice it's not, because I don't have any antivirus or firewall software which could be blocking this.

I don't know if you already solved the problem, but I'm also an openSUSE Leap user, and I also had this problem, I opened YaST > Firewall > I disabled "Start at boot" and restarted the system. It worked for me.

**Comment by Gunstick** [comment]
Problem still exists, so I installed with **betterdiscordctl install**

**Comment by KarsonTheFoxx** [comment]
I had this issue too, just re-launched the installer

**Comment by James-Livesey** [comment]
I got around this by killing Discord using `killall Discord` or `killall DiscordPTB`. I then re-launched the installer and it managed to install just fine. I think the download step might also be trying to kill the Discord process but is not able to, so killing it manually works.

**Comment by UnderscoreAngel** [comment]
apparently this happens because of issues with files and stuff, like creating directories, first is uninstalling betterdiscord if you tried installing it, then tweak the permissions of the installer allowing to run it as an executable and also allowing it to read and write, that's what worked with me, hope i help someone.

**Comment by tschallacka** [comment]
What fixed it for me was opening the console at the agreement window., check the disable cache checkbox under network, then continue the normal installer. Keeping the console open all the time.
